README.md

```diff
@@ -28,14 +28,222 @@ In addition, that format would include optional **signing** of the resources, wh
+   be ignored.
+4. The offset of the [main content](#main-content) of the package.
+5. The metadata sections promised above.
+6. A sequence of HTTP responses constituting the main content of the package.
+   Mismatches with the manifest usually cause the package to fail to parse, as
+   described in the [manifest](#manifest) section, but mismatches with the index
+   might not.
+7. The length of the package, repeated from earlier. This allows tools to
+   operate on the package even if it's been prepended with a self-extracting
+   executable.
+8. The same magic number: `F0 9F 8C 90 F0 9F 93 A6` (🌐📦 in UTF-8).
+
+The MIME type of this format is `application/package`. TODO: Is that right?
+
+Note that this top-level information is *not signed*, and so can't be trusted.
+Only information in the manifest and below can be trusted.
+
+The manifest and certificates are used to sign the contents of the package. The
+index is used when the package is stored on disk, to find resources quickly. And
+the start page should be loaded when a UA is navigated to the package itself.
+
+### Index
+
+The index is optional, and can be added by a client after the package is
+downloaded.
+
+It contains a series of lines (terminated by `\r\n`) of the form
 
-We also propose to remove the [fragment-based URL schema](https://w3ctag.github.io/packaging-on-the-web/#fragment-identifiers) from the spec as it is not clear what would be the use case supporting it.
+```
+URL offset length
+```
+
+Where offset and length are 64-bit integers, and the URLs are absolute. Offsets
+are interpreted relative to the first byte of the package. All resources in the
+package should be included in the index, even if they come from a different
+origin.
+
+TODO: Provide base URLs, maybe as INI-format sections.
+
+```
+https://example.org/index.html 7000 500
+https://example.org/img.png 7500 1000
+...
+```
+
+### Certificates
+
+The certificates block contains
+an
+[application/pkcs7-mime; smime-type=certs-only](https://tools.ietf.org/html/rfc5751) resource
+holding enough [X.509](https://tools.ietf.org/html/rfc5280) certificates that
+chains can be built from the certificates that sign the top-level
+and [sub-package](#sub-packages) manifests to root certificates recognized by
+the client.
+
+Requirements on the
+certificates' [Key Usage](https://tools.ietf.org/html/rfc5280#section-4.2.1.3)
+and [Extended Key Usage](https://tools.ietf.org/html/rfc5280#section-4.2.1.12)
+are TBD. It may or may not be important to prevent TLS serving certificates from
+being used to sign packages, in order to prevent cross-protocol attacks.
+
+### Manifest
+
+A package's manifest lists all of the resources included in that package, and
+any sub-packages the package depends on. It starts with at least the
+`Content-Type`, `Content-Location`, `Content-Length`, and `Date` headers, and
+must include one or more [signatures](#signatures).
```

Comment:
Signatures are also optional. Not-signed packages may have a lot of utility since they can be produced locally and shared easily. The limits on contents are obvious, but for many use cases (packaging a movie that has its own DRM schema embedded, or a static snapshot of a page) the signing is not needed.